tests/jballerina-unit-test/src/test/resources/test-src/statements/retrystmt/retry_stmt_with_on_fail.bal

```diff
@@ -95,6 +103,22 @@ function retryErrorWithFail() returns string|error {
     }
 }
 
+function retryErrorWithFailWithoutVariable() returns string|error {
+    string str = "start";
+    int count = 0;
+    retry<MyRetryManager> (3) {
+        count = count+1;
+        if (count < 3) {
+            str += (" attempt " + count.toString() + ":error,");
+            fail trxError();
+        }
+        str += (" attempt "+ count.toString() + ":result returned end.");
+        return str;
+    } on fail {
```

Comment:
Can we update this test or include another one testing control being transferred to the on-fail clause?